A32NX/html_ui/Pages/A32NX_Core/A32NX_FWC.js

```diff
@@ -286,4 +309,81 @@ class A32NX_FWC {
         this.ldgMemo = showInApproach || invalidRadioMemo || this.flightPhase === 8 || this.flightPhase === 7;
         SimVar.SetSimVarValue("L:A32NX_FWC_LDGMEMO", "Bool", this.ldgMemo);
     }
+
+    altitudeWarning(_deltaTime) {
+        const indicatedAltitude = Simplane.getAltitude();
+        const reset = SimVar.GetSimVarValue("L:A32NX_ALT_DEVIATION_SHORT", "Bool");
+        if (reset = true) {
+            SimVar.SetSimVarValue("L:A32NX_ALT_DEVIATION_SHORT", "Bool", false);
+        }
+
+        // Exit when:
+        // - Landing gear down
+        // - Glide slope captured
+        const landingGearIsDown = !SimVar.GetSimVarValue("IS GEAR RETRACTABLE", "Boolean") || SimVar.GetSimVarValue("GEAR HANDLE POSITION", "Boolean");
+        const glideSlopeCaptured = SimVar.GetSimVarValue("L:GLIDE_SLOPE_CAPTURED", "bool") === 1;
+        if (landingGearIsDown || glideSlopeCaptured) {
+            return;
+        }
+
+        const currentAltitudeConstraint = SimVar.GetSimVarValue("L:A32NX_AP_CSTN_ALT", "feet");
+        // Use the constraint altitude if provided otherwise use selected altitude lock value
+        const targetAltitude = currentAltitudeConstraint && !this.hasAltitudeConstraint() ? currentAltitudeConstraint : Simplane.getAutoPilotSelectedAltitudeLockValue();
+
+        // Exit when selected altitude is being changed
+        if (this.previousTargetAltitude !== targetAltitude) {
+            this.previousTargetAltitude = targetAltitude;
+            this._wasBellowThreshold = false;
+            this._wasAboveThreshold = false;
+            this._wasInRange = false;
+            return;
+        }
+
+        if (this.warningPressed === true) {
+            this._wasBellowThreshold = false;
+            this._wasAboveThreshold = false;
+            this._wasInRange = false;
+            SimVar.SetSimVarValue("L:A32NX_ALT_DEVIATION", "Bool", false);
+            return;
```

Comment:
Warning pressed condition should be moved before landing gear/glide slope conditions. Otherwise the continuous deviation sound cannot be inhibited by pressing the buttons if landing gear is down or glide slope is captured.